### PR TITLE
Clean test_multidevice_sharding.cpp.

### PR DESCRIPTION
**Pull request by wujingyue**
1. Improved naming.
2. Added more tests for broadcast.

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
!build

File: tests/cpp/test_multidevice_sharding.cpp [hunk 1]
```diff
@@ -16,67 +16,68 @@
 namespace nvfuser {
 
 // params: concrete vs symbolic input, sharded axis
-class MultideviceShardingTest
+class MultiDevicePointwiseTest
```
Comment (cowanmeg) [inline tests/cpp/test_multidevice_sharding.cpp:19]: This test isn't a pointwise test though?

Comment (wujingyue) [reply tests/cpp/test_multidevice_sharding.cpp]: Correct. Renamed to MultiDeviceReductionTest.

**Comment by wujingyue** [comment]
!build

**Review by cowanmeg — APPROVED** [review]

**Pull request merged**